app/Observers/EventAchievementObserver.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Observers;
+
+use App\Models\Achievement;
+use App\Models\EventAchievement;
+use App\Models\PlayerAchievement;
+use App\Platform\Jobs\UnlockPlayerAchievementJob;
+
+class EventAchievementObserver
+{
+    public function saved(EventAchievement $eventAchievement): void
+    {
+        // if the source achievement or active period changed, copy the
+        // relevant data to the event achievement
+        $needsCopy = $eventAchievement->wasRecentlyCreated
+            || $eventAchievement->wasChanged([
+                'source_achievement_id', 'active_from', 'active_until',
+            ]);
+
+        if ($needsCopy) {
+            $eventAchievement->loadMissing(['achievement', 'sourceAchievement']);
+
+            $achievement = $eventAchievement->achievement;
+            $sourceAchievement = $eventAchievement->sourceAchievement;
+
+            if ($achievement && $sourceAchievement) {
+                // make the event achievement look like the source achievement
+                $achievement->title = $sourceAchievement->title;
+                $achievement->description = $sourceAchievement->description;
+                $achievement->BadgeName = $sourceAchievement->BadgeName;
```

Comment:
What happens if any of these values change in the source achievement after the event achievement has been set? Should there be something to resync the event achievement? Or would it be better to almost treat the event achievement like a "pointer" of sorts to the source?